src/main/java/org/eclipse/yasson/model/PropertyModel.java

```diff
@@ -136,18 +133,40 @@ private PropertyCustomization introspectCustomization(Property property, JsonbCo
         final AnnotationIntrospector introspector = jsonbContext.getAnnotationIntrospector();
         final PropertyCustomizationBuilder builder = new PropertyCustomizationBuilder();
         //drop all other annotations for transient properties
-        if (introspector.isTransient(property)) {
-            builder.setJsonbTransient(true);
-            return builder.buildPropertyCustomization();
+        EnumSet<AnnotationTarget> transientInfo = introspector.getJsonbTransientCategorized(property);
+        if (transientInfo.size() != 0) {
+            if(transientInfo.contains(AnnotationTarget.GETTER)) {
+                builder.setReadTransient(true);
+            }
+            if (transientInfo.contains(AnnotationTarget.SETTER)) {
+                builder.setWriteTransient(true);
+            }
+            if (transientInfo.contains(AnnotationTarget.PROPERTY)) {
+                if(!transientInfo.contains(AnnotationTarget.GETTER)){
+                    builder.setReadTransient(true);
+                }
+                if(!transientInfo.contains(AnnotationTarget.SETTER)){
+                    builder.setWriteTransient(true);
+                }
+            }
         }
-        builder.setJsonReadName(introspector.getJsonbPropertyJsonReadName(property));
-        builder.setJsonWriteName(introspector.getJsonbPropertyJsonWriteName(property));
-        builder.setNillable(classModel.getClassCustomization().isNillable() || introspector.isPropertyNillable(property));
+
+        if(!builder.isReadTransient()){
```

Comment:
Based on our conversation on #17, the behaviour is as follow:

The exception would be risen only if we have defined colliding annotations on same scope (both on property, both on getter or both on setter). Also if @JsonbTransient is defined for property, the exception would be risen if colliding annotation has been defined for its getter or setter. Other conditions are acceptable. 

- If a property is annotated with @JsonbTransient
  - then none of other annotations could be defined for either the property or its getter and setter
- if a getter is annotation with @JsonbTransient
  - then none of other annotations could be defined for the getter, but it would be allowed for setter and property
- if a setter is annotation with @JsonbTransient
  - then none of other annotations could be defined for the setter, but it would be allowed for getter and property